### PR TITLE
add token reduction to ConversationalRetrievalChain

### PR DESCRIPTION
**Pull request by nkov**
This worked for me, but I'm not sure if its the right way to approach something like this, so I'm open to suggestions.

Adds class property`max_tokens_limit: Optional[int]` to the `ConversationalRetrievalChain`. The code is basically copied from [`RetreivalQAWithSourcesChain`](https://github.com/nkov/langchain/blob/46d141c6cb6c0fdebb308336d8ae140d8368945a/langchain/chains/qa_with_sources/retrieval.py#L24)

### CODE REVIEW

File: langchain/chains/conversational_retrieval/base.py [hunk 1]
```diff
@@ -116,9 +117,35 @@ class ConversationalRetrievalChain(BaseConversationalRetrievalChain, BaseModel):
     """Chain for chatting with an index."""
 
     retriever: BaseRetriever
+    """Index to connect to."""
+    reduce_k_below_max_tokens: bool = False
+    """Reduce the number of results to return from store based on tokens limit"""
+    max_tokens_limit: int = 3375
```
Comment (hwchase17) [inline langchain/chains/conversational_retrieval/base.py:123]: can we just make this `Optional[int] = None` and if its not None then use it? or is there some downside to that im not thinking of?

Comment (hwchase17) [reply langchain/chains/conversational_retrieval/base.py:123]: imo the fewer nobs to turn the better. can leave in other place for backwards compatability but this is better goig forward. happy to change as well if you dont have time today

thanks for adding this!

Comment (nkov) [reply langchain/chains/conversational_retrieval/base.py]: happy to make that change

**Review by hwchase17 — APPROVED** [review]
awesome, thanks!

**Pull request merged**